### PR TITLE
[ISSUE-805] Upgrade from 1.0.x to 1.1.x

### LINKED ISSUE #805: Default values for fields in CRD
There is a problem with `helm upgrade ...`  and OpenAPI validation
There are new fields between CRD versions and those fields must have default values

### PR DESCRIPTION
**Pull request by rkiselenko**
### Resolves #805

Changes described in this pull [#121](https://github.com/dell/csi-baremetal-operator/pull/121)
## PR checklist
- [x] Add link to the issue
- [x] Choose Project
- [x] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

### CODE REVIEW

**Review by misanthropicat — APPROVED** [review]

**Review by mitsaa — APPROVED** [review]

**Review by safronovD — APPROVED** [review]

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -271,5 +271,9 @@ kubectl get pvc
 
 TODO - add information about CI after https://github.com/dell/csi-baremetal/issues/562
 
+## New fields
```
Comment (mishoyama) [inline docs/CONTRIBUTING.md:274]: `New fields` sound confusing to me. Please add something like `Adding new fields to the Kubernetes Custom Resource Definitions`

**Review by mishoyama — CHANGES_REQUESTED** [review]

**Review by mishoyama — APPROVED** [review]

File: docs/proposals/crd-lifecycle-of-crd.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      name: "{{ .Release.Name }}"
+      labels:
+        app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+        app.kubernetes.io/instance: {{ .Release.Name | quote }}
+        helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: pre-upgrade-job
+        image: "alpine:3.3"
```
Comment (safronovD) [inline docs/proposals/crd-lifecycle-of-crd.md:77]: Do we need to create own image with the pre-loaded CRDs and then patch them in command section? Or will it be Go app?

Comment (rkiselenko) [reply docs/proposals/crd-lifecycle-of-crd.md:77]: This is an example.

File: docs/proposals/crd-lifecycle-of-crd.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+## Implementation
+
+Hooks are just Kubernetes manifest files with special annotations in the metadata section. Because they are template files, you can use all of the normal template features, including reading `.Values`, `.Release`, and `.Template`.
+
+ - `pre-upgrade` - 	Executes on an upgrade request after templates are rendered, but before any resources are updated
+
+This template for example, stored in `templates/pre-upgrade-job.yaml`, declares a job to be run on pre-upgrade
+
+```yaml
```
Comment (mishoyama) [inline docs/proposals/crd-lifecycle-of-crd.md:49]: @rkiselenko do you have example of CRD upgrade yaml?

**Review by yurkov-anton — APPROVED** [review]

File: docs/proposals/crd-lifecycle-of-crd.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+## Background
+
+Helm provides a hook mechanism to allow chart developers to intervene at certain points in a release's life cycle.
+Hooks work like regular templates, but they have special annotations that cause Helm to utilize them differently.
+The main focus of these proposal is to turn the CRD into a managed part of the chart and not just an item that is added at install time and then forgotten.
+If the CRD was managed as part of the chart it would fix a lot of confusion **realted to CRDs not being updated when there are changes** and not being deleted with the chart
+
+## Proposal
+
+During helm upgrade upgrade the new CRDs will be applied manually with a shell command running `kubectl patch ...` of the crd.
```
Comment (safronovD) [inline docs/proposals/crd-lifecycle-of-crd.md:38]: Could you please add info about the automated way for upgrade process using helm-hooks?

**Review by mishoyama — APPROVED** [review]

File: docs/crd-lifecycle-of-crd.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+CRDs are never deleted. Deleting a CRD automatically deletes all of the CRD's contents across all namespaces in the cluster. Consequently, Helm will not delete CRDs.
+
+**Operators who want to upgrade or delete CRDs are encouraged to do this manually and with great care**.
+
+## Background
+
+Helm provides a hook mechanism to allow chart developers to intervene at certain points in a release's life cycle.
+Hooks work like regular templates, but they have special annotations that cause Helm to utilize them differently.
+The main focus of these proposal is to turn the CRD into a managed part of the chart and not just an item that is added at install time and then forgotten.
+If the CRD was managed as part of the chart it would fix a lot of confusion **realted to CRDs not being updated when there are changes** and not being deleted with the chart
```
Comment (misanthropicat) [inline docs/crd-lifecycle-of-crd.md:34]: Typo in 'related' word.

**Pull request merged**